feature-utils/poly-analysis/src/utils/analysis-util.js

```diff
@@ -16,3 +16,28 @@ export async function relevantZipEntries(zipFile) {
             !entry.path.includes("__MACOSX")
     );
 }
+
+export function groupActivitiesByTime(activityDates) {
+    let groupedActivities = { total: 0, values: {} };
```

Comment:
As usual, I would feel more comfortable with this being an encapsulated object, but not a big deal.